Assets/Scripts/MapManager.cs

```diff
@@ -94,6 +95,25 @@ IEnumerator LoadLevel()
             string[] textLines = www.downloadHandler.text.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
             GenerateMap(textLines);
             transform.GetComponent<FireController>().enabled = true;
+            CreateWallAroundMap();
+        }
+    }
+
+    private void CreateWallAroundMap()
+    {
+        for(int i = 0; i < mapSizeX; i++)
+        {
+            Vector3 spawnLocation = new Vector3(i, 0, mapSizeY);
+            Instantiate(borderWallPrefab, spawnLocation, Quaternion.identity, gameObject.transform);
+            spawnLocation = new Vector3(i, 0, -1);
+            Instantiate(borderWallPrefab, spawnLocation, Quaternion.identity, gameObject.transform);
+        }
+        for (int i = 0; i < mapSizeY; i++)
+        {
+            Vector3 spawnLocation = new Vector3(mapSizeX, 0, i);
+            Instantiate(borderWallPrefab, spawnLocation, Quaternion.identity, gameObject.transform);
+            spawnLocation = new Vector3(-1, 0, i);
+            Instantiate(borderWallPrefab, spawnLocation, Quaternion.identity, gameObject.transform);
         }
     }
```

Comment:
please separate both parts to one function with arguments, cause they're practically identical

Reply:
the only common part is the instantiate?